### DISCUSSION TITLE
AuthApiError: Signup requires a valid password

### DISCUSSION THREAD

**Discussion by mrtnrs** [supabase/supabase#17597] [Questions] [answered]
I need some guidance here. I have a basic email/pwd register/login setup in my app using:

`const { user, error } = await supabase.auth.signUp({ email, password });`

This worked fine. After a break, the only thing that changed is the localhost port, now when registering a new user I get:

> AuthApiError: Signup requires a valid password

The passwords I'm trying are sufficiently long as set up in the settings. 

I don't understand, I believe nothing changed in my setup + the password is fine. What's going on? Thanks for the help.

**Comment by GaryAustin1** [comment]
Have you consoled logged password right at the call?  Can you see the network request in the browser developer console?

**Reply by mrtnrs** [reply]
Yes, first thing I did was log email and pwd the line above auth.signUp and it logs correctly. I can see the network request.

Weirdly enough also logging in doesn't work, which returns a similar error (Error signing in: Invalid login credentials):

...
{error: 'invalid_grant', error_description: 'Invalid login credentials'}
status: 400
url: "https://myurl.supabase.co/auth/v1/token?grant_type=password"
...

I thought maybe somehow my url changed in supabase but it's still the same and the post request uses my url correctly.

Only changes I made since coming back from vacation:

- different localhost port
- updated supabase-js to latest version coming from I believe 2.23.0 to 2.33.2
- removed some irrelevant firebase dependencies

I'm at a loss on how to solve this because the problem is elsewhere than the actual credentials... First thing tomorrow I'll try downgrading supabase to see if that helps.

**Reply by GaryAustin1** [reply]
I was looking around and there is an old issue that the message is used for both email bad and password bad.  https://github.com/supabase/gotrue/issues/323

**Reply by mrtnrs** [reply]
I've been loggin in with these credentials for months, there's nothing wrong with them...

The request:

`Object
callbackId: "24124043"
methodName: "request"
options: 
data: ReadableStream {locked: false}
dataType: "json"
headers: 
{apikey: 'api key here', authorization: 'Bearer api key here', content-type: 'application/json;charset=UTF-8', x-client-info: 'supabase-js/2.23.0'}
method: "POST"
url: "https://uyjnwwgxxgxfnhlcdvjn.supabase.co/auth/v1/token?grant_type=password"
[[Prototype]]: Object
pluginId: "CapacitorHttp"`

Returns:

`data: {error: 'invalid_grant', error_description: 'Invalid login credentials'}
error: true
headers: {Access-Control-Allow-Origin: '*', alt-svc: 'h3=":443"; ma=86400', CF-Cache-Status: 'DYNAMIC', CF-Ray: '80a38cb9cc7eb9a5-BRU', Connection: 'keep-alive', …}
status: 400
url: "https://uyjnwwgxxgxfnhlcdvjn.supabase.co/auth/v1/token?grant_type=password"`

I downgraded supabase-js but the problem persists. I can see the user in the db, so the user is still there. Somehow any connection attempt to my user db, be it logging in or registering a new user fails, it's as if I'm somehow connected to a different db but everything seems fine...

**Reply by GaryAustin1** [reply]
signUp should not care what your old credentials (password/email) were.  Should just care that they are valid.  What does your `data:ReadableStream{locked:false}` mean?   What is your body for the Post?

**Reply by mrtnrs** [reply, marked as answer]
Found out this was related to a bug in CapacitorHttp, which led to the request being empty... So in the end the error returned by Supabase was correct.

https://github.com/ionic-team/capacitor/issues/6645

Updating capacitor to the latest version fixed it 🥲 Rip my last 6 hours but thanks for the support.
